Update icon annotation references in the documentation

adamscott opened the pull request:
Due to godotengine/godot#67774, the icon annotation is now required to be on the top of the file, as annotations must come before the thing that they are annotating.

This PR updates the doc to match the new engine behavior about the icon annotation.

Fixes #6629.

adamscott commented on classes/class_@gdscript.rst:
```diff
@@ -339,11 +339,11 @@ See also :ref:`@GlobalScope.PROPERTY_USAGE_GROUP<class_@GlobalScope_constant_PRO
     @export var number = 3
     @export var string = ""
-    
+
```
There's some `.editorconfig` trailing whitespace trim in my PR.

EnhancedJunk commented:
Hey, I noticed there is more example code to be updated around line 1316 as well.
Also, there seems to be a typo with a randomly placed `the` in the note: `annotations describe the their subject`

adamscott commented:
Thanks @EnhancedJunk!

YuriSizov commented:
The class reference should not be edited here, it should be edited in the XML on the main repo.

adamscott commented:
> The class reference should not be edited here, it should be edited in the XML on the main repo.

Removed the class reference edits. Added them in godotengine/godot#71620.

skyace65 reviewed (approved):
Looks good, thanks!
